xc/device.ml

```diff
@@ -772,6 +775,49 @@ module Vif = struct
 
 end
 
+(****************************************************************************************)
+(** Network SR-IOV VFs:                                                                 *)
+module NetSriovVf = struct
+
+  let add  ~xs ~devid ~mac ?mtu ?(rate=None) ?(backend_domid=0) ?(other_config=[]) 
+      ~pci ~vlan ~carrier ?(extra_xenserver_keys=[]) (task: Xenops_task.task_handle) domid =
+    let open Xcp_pci in
+    let vlan_str = match vlan with None -> "none" | Some vlan -> sprintf "%Ld" vlan in
+    let rate_str = match rate with None -> "none" | Some (a, b) -> sprintf "(%Ld,%Ld)" a b in
+    debug "Device.NetSriovVf.add domid=%d devid=%d pci=%s vlan=%s mac=%s carrier=%b \
+           rate=%s other_config=[%s] extra_xenserver_keys=[%s]" 
+      domid devid (string_of_address pci)  vlan_str mac carrier rate_str
+      (String.concat "; " (List.map (fun (k, v) -> k ^ "=" ^ v) other_config))
+      (String.concat "; " (List.map (fun (k, v) -> k ^ "=" ^ v) extra_xenserver_keys));
+
+    let frontend = { domid = domid; kind = NetSriovVf; devid = devid } in
+    let backend = { domid = backend_domid; kind = NetSriovVf; devid = devid } in
+    let device = { backend = backend; frontend = frontend } in
+
+    Generic.add_device ~xs device [] [] [] (extra_xenserver_keys @ other_config);
```

Comment:
Just to double check, the informations in `device` are enough and we don't need any `back` and `front` list for the `Generic.add_device` call. Right?

Reply:
Yes.
`backend` and `frontend` are for PV drivers only.
In the case of netowrk SR-IOV, instead of PV drivers, PCI-passthrough is the underlying function.
So `back` and `front` are both empty here.